lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -591,6 +607,32 @@ class ReportUIFeatures {
     const popup = window.open(url, windowName);
   }
 
+  /**
+   * Opens a new tab to an external page and sends data via base64 encoded url params.
+   * @param {{lhr: LH.Result} | LH.Treemap.Options} data
+   * @param {string} url_
+   * @param {string} windowName
+   * @protected
+   */
+  static openTabWithUrlData(data, url_, windowName) {
+    const url = new URL(url_);
+    url.hash = toBinary(JSON.stringify(data));
+
+    // The popup's window.name is keyed by version+url+fetchTime, so we reuse/select tabs correctly
+    window.open(url.toString(), windowName);
+
+    /**
+     * @param {string} string
+     */
+    function toBinary(string) {
+      const codeUnits = new Uint16Array(string.length);
+      for (let i = 0; i < codeUnits.length; i++) {
+        codeUnits[i] = string.charCodeAt(i);
+      }
+      return btoa(String.fromCharCode(...new Uint8Array(codeUnits.buffer)));
```

Comment:
For here and in `fromBinary`, I think we need to avoid spread. There have been optimizations to improve the perf of spread in function calls (though I don't know the cross browser story off hand), but large payloads will still hit the argument length limit and throw an error

Reply:
> For here and in `fromBinary`, I think we need to avoid spread. There have been optimizations to improve the perf of spread in function calls (though I don't know the cross browser story off hand), but large payloads will still hit the argument length limit and throw an error

`String.fromCharCode(...new Uint8Array(125183))` throws on my machine in Chrome, for instance, but it's 2**16 + 1 in Safari.

Reply:
Wow even `String.fromCharCode.apply(null, new Uint8Array(125183))` breaks.

Reply:
don't we have to be careful about unicode here? don't see how to do this safely in chunks..

Reply:
should we doing codePointAt ??